src/webui/www/private/scripts/dynamicTable.js

```diff
@@ -2650,8 +2658,15 @@ window.qBittorrent.DynamicTable ??= (() => {
 
             this._updateNodeCollapseIcon(node, shouldCollapse);
 
-            for (const child of node.children)
-                this._updateNodeVisibility(child, shouldCollapse);
+            this._updateNodeChildVisibility(node, shouldCollapse);
+        },
+
+        _updateNodeChildVisibility: function(node, shouldHide) {
+            for (const child of node.children) {
+                this._updateNodeVisibility(child, shouldHide);
+                if (child.isFolder)
+                    this._updateNodeChildVisibility(child, shouldHide ? shouldHide : this.isCollapsed(child.rowId));
```

Comment:
Please don't use recursion. Take a look at https://github.com/qbittorrent/qBittorrent/commit/0187f19f60ed685902480e115400253f2eae85d6 to convert them.